From 76c39627ebc114913e532474f0c4e98c4fed3d14 Mon Sep 17 00:00:00 2001
From: Michael Prokop <mprokop@sipwise.com>
Date: Tue, 2 Sep 2025 12:14:35 +0200
Subject: [PATCH] Add support for Debian 13 (trixie)

Debian v13 AKA trixie was released on 2025-08-09 as Debian's new stable
release.
---
 manifests/params.pp       |  3 +++
 metadata.json             |  3 ++-
 spec/classes/java_spec.rb | 18 ++++++++++++++++++
 3 files changed, 23 insertions(+), 1 deletion(-)

diff --git a/manifests/params.pp b/manifests/params.pp
index 708d3e6f..fa352809 100644
--- a/manifests/params.pp
+++ b/manifests/params.pp
@@ -58,6 +58,9 @@
         default   => $facts['os']['architecture']
       }
       case $facts['os']['release']['major'] {
+        '13': {
+          $openjdk = 21
+        }
         '12', '24.04': {
           $openjdk = 17
         }
diff --git a/metadata.json b/metadata.json
index 40bd8099..d1866765 100644
--- a/metadata.json
+++ b/metadata.json
@@ -44,7 +44,8 @@
       "operatingsystemrelease": [
         "10",
         "11",
-        "12"
+        "12",
+        "13"
       ]
     },
     {
diff --git a/spec/classes/java_spec.rb b/spec/classes/java_spec.rb
index 90efa9f5..dd51a098 100644
--- a/spec/classes/java_spec.rb
+++ b/spec/classes/java_spec.rb
@@ -46,6 +46,24 @@
     end
   end
 
+  context 'on Debian Trixie (13)' do
+    let(:facts) { { os: { family: 'Debian', name: 'Debian', lsb: { distcodename: 'trixie' }, release: { major: '13' }, architecture: 'amd64' } } }
+
+    context 'when selecting jdk' do
+      let(:params) { { 'distribution' => 'jdk' } }
+
+      it { is_expected.to contain_package('java').with_name('openjdk-21-jdk') }
+      it { is_expected.to contain_file_line('java-home-environment').with_line('JAVA_HOME=/usr/lib/jvm/java-1.21.0-openjdk-amd64/') }
+    end
+
+    context 'when selecting jre' do
+      let(:params) { { 'distribution' => 'jre' } }
+
+      it { is_expected.to contain_package('java').with_name('openjdk-21-jre-headless') }
+      it { is_expected.to contain_file_line('java-home-environment').with_line('JAVA_HOME=/usr/lib/jvm/java-1.21.0-openjdk-amd64/') }
+    end
+  end
+
   context 'on Ubuntu Bionic (18.04)' do
     let(:facts) { { os: { family: 'Debian', name: 'Ubuntu', lsb: { distcodename: 'bionic' }, release: { major: '18.04' }, architecture: 'amd64' } } }
 
